### PR TITLE
[bazel] Use the new rules for building ROMs 

### PR DESCRIPTION
**Pull request by cfrantz**
1. Rewrite `sw/device/silicon_creator/rom/BUILD` to use the new binary rules.
2. Add an `output_groups` rule that can select several output groups from an OutputGroupInfo provider and emit a DefaultInfo provider with all of the requested files.
3. Rework the packaging of the `hello_world` and ROM binaries to use the `output_groups` rule.

### CODE REVIEW

**Comment by cfrantz** [comment]
Fixes #19922.

**Comment by pamaury** [comment]
Unfortunately, a few tests in rom e2e directly refers to `rom_with_fake_keys_fpga_cw310.elf` which is now inaccessible. You might to change those to `rom_with_fake_keys_fpga_cw310_elf` as well?

File: sw/device/examples/hello_world/BUILD [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 load("//rules/opentitan:defs.bzl", "opentitan_binary")
-load("//rules:exclude_files.bzl", "exclude_files")
+load("//rules:files.bzl", "exclude_files")
```
Comment (moidx) [inline sw/device/examples/hello_world/BUILD:6]: nit: for a follow up PR if applicable. Should we remove this load line? `exclude_files` is no longer used in this file.

File: sw/device/silicon_creator/rom/BUILD [hunk 2]
```diff
@@ -159,65 +154,68 @@ cc_library(
...
 )
 
-_ROM_DEVICES = [
-    "sim_dv",
-    "sim_verilator",
-    "fpga_cw310",
-]
-
-[
-    elf_to_scrambled_rom_vmem(
```
Comment (moidx) [inline sw/device/silicon_creator/rom/BUILD:195]: Is this now done by default for each target using `opentitan_binary`? `sim_dv` probably needs it.

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/rom/BUILD (hunk 2 above)
Comment (cfrantz) [reply sw/device/silicon_creator/rom/BUILD:195]: Yes. For sim_dv it is done here: https://cs.opensource.google/opentitan/opentitan/+/master:rules/opentitan/sim_dv.bzl;l=45-55

Similar bits of code exist for sim_verilator and fpga_cw310, each producing the 39-bit scrambled vmem files.

File: sw/device/examples/hello_world/BUILD (hunk 1 above)
Comment (cfrantz) [reply sw/device/examples/hello_world/BUILD:6]: yes.

**Pull request merged**